react/features/prejoin/components/Prejoin.native.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useCallback, useLayoutEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Text, View, TouchableOpacity, TextInput, Platform } from 'react-native';
+import { useDispatch, useSelector, useStore } from 'react-redux';
+
+import { connect } from '../../base/connection/actions.native';
+import { IconClose } from '../../base/icons';
+import JitsiScreen from '../../base/modal/components/JitsiScreen';
+import { getLocalParticipant } from '../../base/participants';
+import { getFieldValue } from '../../base/react';
+import { ASPECT_RATIO_NARROW } from '../../base/responsive-ui';
+import { updateSettings } from '../../base/settings';
+import { LargeVideo } from '../../large-video/components';
+import HeaderNavigationButton from '../../mobile/navigation/components/HeaderNavigationButton';
+import { goBackToRoot, navigateRoot } from '../../mobile/navigation/rootNavigationContainerRef';
+import { screen } from '../../mobile/navigation/routes';
+import AudioMuteButton from '../../toolbox/components/AudioMuteButton';
+import VideoMuteButton from '../../toolbox/components/VideoMuteButton';
+
+import styles from './styles';
+
+
+interface Props {
+    navigation: any;
+}
+
+const Prejoin: ({ navigation }: Props) => JSX.Element = ({ navigation }: Props) => {
+    const dispatch = useDispatch();
+    const { t } = useTranslation();
+    const aspectRatio = useSelector(
+        (state: any) => state['features/base/responsive-ui']?.aspectRatio
+    );
+    const store = useStore();
+    const state = store.getState();
+    const localParticipant = useSelector(state => getLocalParticipant(state));
+    const participantName = localParticipant?.name;
+    const [ displayName, setDisplayName ]
+        = useState(participantName || '');
+    const onChangeDisplayName = useCallback(event => {
+        const fieldValue = getFieldValue(event);
+
+        setDisplayName(fieldValue);
+        dispatch(updateSettings({
+            displayName: fieldValue
+        }));
+    }, [ displayName ]);
+
+    const onJoin = useCallback(() => {
+        dispatch(connect());
+        navigateRoot(screen.conference.root);
+    }, [ dispatch ]);
+
+    const goBack = useCallback(() => {
+        goBackToRoot(state, dispatch);
```

Comment:
Since we might create tracks, we need them destroyed. Hence, you need to `appNavigate` here too.